pkg/agent/multicluster/mc_route_controller.go

```diff
@@ -362,7 +367,10 @@ func (c *MCRouteController) addMCFlowsForSingleCIImp(activeGW *mcv1alpha1.Gatewa
 
 	klog.InfoS("Adding/updating remote Gateway Node flows for Multi-cluster", "gateway", klog.KObj(activeGW),
 		"node", c.nodeConfig.Name, "peer", tunnelPeerIPToRemoteGW)
-	allCIDRs := append([]string{ciImport.Spec.ServiceCIDR}, ciImport.Spec.PodCIDRs...)
+	allCIDRs := []string{ciImport.Spec.ServiceCIDR}
+	if c.enablePodToPodConnectivity {
```

Comment:
When Pod connectivity is not enabled, do we have extra routines to disable? If it is just about adding remote Pod CIDRs for Gateway routing/tunneling, seems not much value to add a new knob? It can already be controlled by Pod CIDRs in MC Controller config right.

Reply:
Yes. In the previous design, we just use Pod CIDRs to control this feature. But the Pod CIDRs are also needed by WireGuard feature, so I think maybe we should add a toggle for this one.

Reply:
Ok. I did not know this. Make sense then.